instaclustr/resource_cluster.go

```diff
@@ -725,6 +709,64 @@ func resourceClusterRead(d *schema.ResourceData, meta interface{}) error{
 	return nil
 }
 
+func getBundlesFromCluster(cluster *Cluster) ([]map[string]interface{}, error) {
+	baseBundle := make(map[string]interface{}, 0)
+	baseBundle["bundle"] = cluster.BundleType
+
+	baseBundleOptions := make(map[string]interface{}, 0)
+	err := mapstructure.Decode(cluster.BundleOption, &baseBundleOptions)
+	if err != nil {
+		return nil, fmt.Errorf("[Error] Error decoding bundles option: %s", err)
+	}
+
+	convertedBundleOptions := convertInterface(baseBundleOptions)
+
+	baseBundle["options"] = convertedBundleOptions
+	baseBundle["version"] = cluster.BundleVersion
+
+	bundles := make([]map[string]interface{}, 0)
+	bundles = append(bundles, baseBundle)
+	if cluster.AddonBundles != nil {
+		bundles = append(bundles, cluster.AddonBundles)
+	}
+
+	return bundles, nil
+}
+
+func convertInterface(data map[string]interface{}) map[string]interface{} {
+	for k, v := range data {
+		// Terraform expects strings for everything
+		// This block iterates through the bundle options map and checks for an interface of a pointer
+		// For each interface{*type} value, it changes: interface{*type} -> *type -> type -> interface{type} -> String
+		// For a non-pointer, it directly formats to a string
+		valueOfV := reflect.ValueOf(v)
+		if valueOfV.Kind() == reflect.Ptr {
+			data[k] = fmt.Sprintf("%v", reflect.Indirect(valueOfV.Elem()).Interface())
+		} else {
+			data[k] = fmt.Sprintf("%v", valueOfV)
+		}
+	}
+
+	return data
+}
+
+func processProvider(d *schema.ResourceData, data map[string]interface{}) (newData map[string]interface{}) {
```

Comment:
Is this only useful for processing the provider? Or could it be used more generally for other string fields?

Reply:
I feel like this is still kind of a hacky solution and we should try to avoid using it